components/logins/src/db.rs

```diff
@@ -645,6 +645,24 @@ impl LoginDb {
         Ok(exists)
     }
 
+    pub fn delete_local(&self, id: &str) -> Result<bool> {
```

Comment:
I think a comment here explaining why these 2 deletions are the "right thing" to do in this scenario and how it causes us to re-sync the remote item (which IIUC, also requires the engine to be "reset" - which I guess is done in the swift layer?)

Reply:
@mhammond Yeah, I just convinced myself those two functions shouldn't be exposed. Let me know if you think `reset` is prefered over my `set_last_sync` usage.

Reply:
`set_last_sync` seems fine here! I agree a single `verify_logins` function seems better than exposing `delete_local` and `set_last_sync` too!